assets/js/googlesitekit/modules/datastore/sharing-settings.test.js

```diff
@@ -0,0 +1,367 @@
+					.setSharedRoles( moduleSlugs[ 1 ], sharedRoles );
+
+				const sharingSettingsWithSharedRoles = moduleSlugs.reduce(
+					( sharingSettingsObj, moduleSlug ) => ( {
+						...sharingSettingsObj,
+						[ moduleSlug ]: {
+							...settingsWithoutRoles[ moduleSlug ],
+							sharedRoles,
+						},
+					} ),
+					{}
+				);
+
+				expect( store.getState().sharingSettings ).toMatchObject( {
+					...state,
+					...sharingSettingsWithSharedRoles,
+				} );
+			} );
+		} );
+
+		describe( 'saveSharingSettings', () => {
+			it( 'does not require any params', () => {
+				expect( async () => {
+					fetchMock.get(
+						/^\/google-site-kit\/v1\/core\/modules\/data\/list/,
+						{ body: FIXTURES, status: 200 }
+					);
+
+					fetchMock.postOnce(
+						/^\/google-site-kit\/v1\/core\/modules\/data\/sharing-settings/,
+						{
+							body: {
+								settings: sharingSettings.settings,
+								newOwnerIDs: {
+									'search-console': 2,
+									'pagespeed-insights': 2,
+								},
+							},
+						}
+					);
+
+					const initialModules = registry
+						.select( CORE_MODULES )
+						.getModules();
+					// The modules info will be its initial value while the modules
+					// info is fetched.
+					expect( initialModules ).toBeUndefined();
+					await untilResolved( registry, CORE_MODULES ).getModules();
+
+					// TODO: Remove the `receiveGetSharingSettings` call and
+					// Add coverage for using `getSharingSettings` selector in 4795.
+					registry
+						.dispatch( CORE_MODULES )
+						.receiveGetSharingSettings( sharingSettings.settings );
+
+					await registry
+						.dispatch( CORE_MODULES )
+						.saveSharingSettings();
+				} ).not.toThrow();
+			} );
+
+			it.each( [
+				[
+					'should',
+					{
+						'search-console': 2,
+						'pagespeed-insights': 2,
+					},
+					2,
+				],
+				[ 'should not', undefined, undefined ],
+			] )(
+				'dispatches a request to save sharing settings and %s dispatch setOwnerID action based on the `newOwnerIDs` availability',
+				async ( _, newOwnerIDs, ownerID ) => {
+					fetchMock.get(
+						/^\/google-site-kit\/v1\/core\/modules\/data\/list/,
+						{
+							body: [
+								...FIXTURES,
+								{
+									slug: 'search-console',
+									name: 'Search Console',
+									storeName: 'modules/search-console',
+								},
+								{
+									slug: 'pagespeed-insights',
+									name: 'PageSpeed Insights',
+									storeName: 'modules/pagespeed-insights',
+								},
+							],
+							status: 200,
+						}
+					);
+
+					fetchMock.get(
+						/^\/google-site-kit\/v1\/core\/site\/data\/settings/,
+						{ body: { setting1: 'value' }, status: 200 }
+					);
+
+					fetchMock.get(
+						/^\/google-site-kit\/v1\/modules\/search-console\/data\/settings/,
+						{ body: { setting1: 'value' }, status: 200 }
+					);
+
+					fetchMock.get(
+						/^\/google-site-kit\/v1\/modules\/pagespeed-insights\/data\/settings/,
+						{ body: { setting1: 'value' }, status: 200 }
+					);
```

Comment:
Same here, no need to use fetchMock for this, just use appropriate `receiveGetSettings` actions to set settings for modules.